pkg/app/utils.go

```diff
@@ -47,27 +49,29 @@ func EnsureAppProjectName(userNSClient v1.NamespaceInterface, ownedProjectID, cl
 		}
 	}
 
-	// move Namespace into a project
 	expectedAppProjectName := fmt.Sprintf("%s:%s", clusterName, ownedProjectID)
-	appProjectName := ""
-	if projectName, ok := deployNamespace.Annotations[nslabels.ProjectIDFieldLabel]; ok {
-		appProjectName = projectName
+	appProjectID := deployNamespace.Annotations[nslabels.ProjectIDFieldLabel]
+	if strings.Contains(settings.SystemNamespaces.Get(), deployNamespace.Name) && appProjectID != "" {
```

Comment:
This check needs to be exact, a contains will have false positives if the namespace is something like `fleet-system` when `cattle-fleet-system` exists in the setting. 
Do the `appProjectID != ""` check first so we don't have to process namespaces if we don't need to.

Reply:
Done.